src/scenechange/mod.rs

```diff
@@ -81,18 +114,17 @@ impl SceneChangeDetector {
   ///
   /// This will gracefully handle the first frame in the video as well.
   #[hawktracer(analyze_next_frame)]
-  pub fn analyze_next_frame<T: Pixel>(
+  pub fn analyze_next_frame(
     &mut self, frame_set: &[Arc<Frame<T>>], input_frameno: u64,
     previous_keyframe: u64,
   ) -> bool {
+    // Use score deque for adaptive threshold for scene cut
+    // Declare score_deque offset based on lookahead  for scene change scores
+
     // Find the distance to the previous keyframe.
     let distance = input_frameno - previous_keyframe;
 
-    if frame_set.len() < 2 {
```

Comment:
why did you remove this check?       

I see a 
``` rust
 self.deque_offset = frame_set.len() - 2;
```

just below.

Reply:
IIRC because scene change could handle 2 frames just fine.
I probably should add that back

Reply:
Added back